tests/ha/test_multi_pod_failure.py

```diff
@@ -2341,3 +2342,386 @@ def test_degraded_mpu_after_kpods_fail(self):
+        LOGGER.info("Listed parts of partial multipart upload: %s", res[1])
+        parts = [ele for ele in parts if ele not in uploading_parts]
+        LOGGER.info("Step 1: Successfully performed partial multipart upload for %s parts out "
+                    "of total %s", uploading_parts, len(parts))
+        LOGGER.info("Shutdown %s (K) data pods one by one and Start multipart upload for "
+                    "5GB object in multiple parts with every data pod shutdown", self.kvalue)
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.POD_NAME_PREFIX)
+        LOGGER.info("Get pod names to be deleted")
+        self.pod_name_list = random.sample(pod_list, self.kvalue)
+        for count, pod_name in enumerate(self.pod_name_list):
+            count += 1
+            LOGGER.info("Step 2: Shutdown %s data pod %s by deleting deployment (unsafe)",
+                        count, pod_name)
+            pod_data = list()
+            pod_data.append(
+                self.node_master_list[0].get_pod_hostname(pod_name=pod_name))  # hostname
+            resp = self.node_master_list[0].delete_deployment(pod_name=pod_name)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_false(resp[0], f"Failed to delete {count} pod {pod_name} by "
+                                               "deleting deployment (unsafe)")
+            pod_data.append(resp[1])  # deployment_backup
+            pod_data.append(resp[2])  # deployment_name
+            self.restore_pod = self.deploy = True
+            self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+            self.pod_dict[pod_name] = pod_data
+            LOGGER.info("Step 2: Deleted %s pod %s by deleting deployment (unsafe)",
+                        count, pod_name)
+
+            LOGGER.info("Step 3: Check cluster status")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 3: Cluster is in degraded state")
+
+            LOGGER.info("Step 4: Check services status that were running on pods which are deleted")
+            hostname = self.pod_dict.get(pod_name)[0]
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                               hostname=hostname)
+            assert_utils.assert_true(resp[0], resp[1])
+            pod_list.remove(pod_name)
+            LOGGER.info("Step 4: Services of pods are in offline state")
+
+            LOGGER.info("Step 5: Check services status on remaining pods %s", pod_list)
+            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], resp)
+            LOGGER.info("Step 5: Services of remaining pods are in online state.")
+
+            uploading_parts = random.sample(parts, 10)
```

Comment:
do we want to check if we have exhausted all parts if we have multiple pod failures?